<launch>
	<!-- =================== SET Camera Status (ON/OFF) =================== -->
	<arg name="statusL" default="True"/>
	<arg name="statusR" default="False"/>
	<!-- =================== SET Argument =================== -->
	<arg name="camera_id_left" value="3" />
	<arg name="camera_id_right" value="2" />
	<arg name="nodelet_manager_name_left" value="bottom_left" />
	<arg name="nodelet_manager_name_right" value="bottom_right" />
	<arg name="camera_name_left" value="/bottom/left" />
	<arg name="camera_name_right" value="/bottom/right" />
	<!-- =================== SET Argument LEFT CAMERA =================== -->
	<arg name="ext_trigger_mode" value="False" />
	<arg name="color_mode" value="bgr8" />
	<arg name="image_width" value="1936" />
	<arg name="image_height" value="1216" />
	<arg name="auto_gain" value="False"/>
	<arg name="master_gain" value="0" />
	<arg name="red_gain" value="100" />
	<arg name="green_gain" value="0" />
	<arg name="blue_gain" value="25" />
	<arg name="auto_exposure" value="False"/>
	<arg name="exposure" value="1"/>
	<arg name="auto_white_balance" value="False"/>
	<arg name="auto_frame_rate" value="True"/>
	<arg name="frame_rate" value="15.0"/>
	<arg name="pixel_clock" value="30"/>
	<arg name="flip_lr" value="False" />
	<!-- =================== LEFT CAMERA =================== -->
	<group if="$(arg statusL)">
		<node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name_left)" args="manager" output="screen" />
		<node pkg="nodelet" type="nodelet" name="ueye_cam_nodelet_$(arg nodelet_manager_name_left)" args="load ueye_cam/ueye_cam_nodelet $(arg nodelet_manager_name_left)">
			<param name="camera_name" type="str" value="$(arg camera_name_left)" />
			<param name="camera_bottomic" type="str" value="image_raw" />
			<param name="camera_id" type="int" value="$(arg camera_id_left)" />
			<!-- <param name="camera_intrinsics_file" type="string" value=" /home/zeabus/catkin_ws/src/src_code/zeabus_vision/zeabus_vision_main/params/bottom.yaml"/> -->
			<param name="ext_trigger_mode" type="bool" value="$(arg ext_trigger_mode)" />
			<param name="color_mode" type="str" value="$(arg color_mode)" />
			<param name="image_width" type="int" value="$(arg image_width)" />
			<param name="image_height" type="int" value="$(arg image_height)" />
			<param name="auto_gain" type="bool" value="$(arg auto_gain)" />
			<param name="master_gain" type="int" value="$(arg master_gain)" />
			<param name="red_gain" type="int" value="$(arg red_gain)" />
			<param name="green_gain" type="int" value="$(arg green_gain)" />
			<param name="blue_gain" type="int" value="$(arg blue_gain)" />
			<param name="auto_exposure" type="bool" value="$(arg auto_exposure)" />
			<param name="exposure" type="int" value="$(arg exposure)" />
			<param name="auto_white_balance" type="bool" value="$(arg auto_white_balance)" />
			<param name="auto_frame_rate" type="bool" value="$(arg auto_frame_rate)" />
			<param name="frame_rate" type="double" value="$(arg frame_rate)" />
			<param name="pixel_clock" type="int" value="$(arg pixel_clock)" />
			<param name="flip_lr" type="bool" value="$(arg flip_lr)" />
		</node>
	</group>
	<!-- =================== RIGHT CAMERA =================== -->
	<group if="$(arg statusR)">
		<node pkg="nodelet" type="nodelet" name="$(arg nodelet_manager_name_right)" args="manager" output="screen" />
		<node pkg="nodelet" type="nodelet" name="ueye_cam_nodelet_$(arg nodelet_manager_name_right)" args="load ueye_cam/ueye_cam_nodelet $(arg nodelet_manager_name_right)">
			<param name="camera_name" type="str" value="$(arg camera_name_right)" />
			<param name="camera_bottomic" type="str" value="image_raw" />
			<param name="camera_id" type="int" value="$(arg camera_id_right)" />
			<param name="ext_trigger_mode" type="bool" value="$(arg ext_trigger_mode)" />
			<param name="color_mode" type="str" value="$(arg color_mode)" />
			<param name="image_width" type="int" value="$(arg image_width)" />
			<param name="image_height" type="int" value="$(arg image_height)" />
			<param name="auto_gain" type="bool" value="$(arg auto_gain)" />
			<param name="master_gain" type="int" value="$(arg master_gain)" />
			<param name="red_gain" type="int" value="$(arg red_gain)" />
			<param name="green_gain" type="int" value="$(arg green_gain)" />
			<param name="blue_gain" type="int" value="$(arg blue_gain)" />
			<param name="auto_exposure" type="bool" value="$(arg auto_exposure)" />
			<param name="exposure" type="int" value="$(arg exposure)" />
			<param name="auto_white_balance" type="bool" value="$(arg auto_white_balance)" />
			<param name="auto_frame_rate" type="bool" value="$(arg auto_frame_rate)" />
			<param name="frame_rate" type="double" value="$(arg frame_rate)" />
			<param name="pixel_clock" type="int" value="$(arg pixel_clock)" />
			<param name="flip_lr" type="bool" value="$(arg flip_lr)" />
		</node>
	</group>
	<arg name="imageType" value="image_raw/compressed"/>
	<node pkg="zeabus_vision" name="auto_exposure_bottom" type="auto_exposure_bottom.py" output="screen">
		<param name="imageTopicL" type="str" value="/bottom/left/$(arg imageType)"/>
		<param name="imageClientL" type="str" value="ueye_cam_nodelet_$(arg nodelet_manager_name_left)/"/>
		<param name="imageTopicR" type="str" value="/bottom/right/$(arg imageType)"/>
		<param name="imageClientR" type="str" value="ueye_cam_nodelet_$(arg nodelet_manager_name_right)/"/>
	</node>
</launch>
